base/gmp.jl

```diff
@@ -458,7 +475,7 @@ powermod(x::Integer, p::Integer, m::BigInt) = powermod(big(x), big(p), m)
 
 function gcdx(a::BigInt, b::BigInt)
     if b == 0 # shortcut this to ensure consistent results with gcdx(a,b)
-        return a < 0 ? (-a,-one(BigInt),zero(BigInt)) : (a,one(BigInt),zero(BigInt))
+        return a < 0 ? (-a,-ONE,ZERO) : (a,ONE,ZERO)
```

Comment:
I would love to see something along the lines of https://github.com/JuliaLang/julia/pull/17015 being merged. But in this case, GMP.jl would have to be rewritten anyways (and this change here would be unnecessary).

Since such a change did not take place, the only way to write fast code for `BigInt`s is to directly call the corresponding gmp functions. And this works pretty well at the moment. But with the changes here, this is more error prone.

While the speedup is negligible, with this change mutating a mutable type can change global variables and consequently the behavior of various functions.